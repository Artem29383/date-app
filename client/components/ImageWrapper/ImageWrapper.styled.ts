import styled from "styled-components";
import {
  borderRadius,
  flexBasis,
  flexShrink,
  height,
  margin,
  maxHeight,
  maxWidth,
  overflow,
  width
} from "styled-system";
import { Colors } from "@types";

export const WrapperImage = styled.div<{
  onClick: (() => void) | undefined;
  onDoubleClick: (() => void) | undefined;
}>`
  width: 100%;
  height: 100%;
  display: flex;
  cursor: ${({ onClick, onDoubleClick }) =>
    onClick || onDoubleClick ? "pointer" : "default"};
  justify-content: center;
  align-items: center;
  background-color: ${Colors.bombay};

  ${flexBasis};
  ${borderRadius};
  ${overflow};
  ${maxHeight};
  ${maxWidth};
  ${width};
  ${height};
  ${margin};
  ${flexShrink};
`;

export const Image = styled.img`
  max-width: 100%;
  height: 100%;
  width: 100%;
  object-fit: cover;
  opacity: 0;

  &.thumb {
    opacity: 1;
    filter: blur(10px);
    transition: opacity 1s ease-in-out;
    position: absolute;
    &.isLoaded {
      opacity: 0;
    }
  }

  &.isLoaded {
    transition: opacity 1s ease-in-out;
    opacity: 1;
  }
`;

export const ImageStatic = styled.img`
  max-width: 100%;
  height: 100%;
  width: 100%;
  object-fit: cover;

  &.thumb {
    filter: blur(10px);
    position: absolute;
  }
`;
