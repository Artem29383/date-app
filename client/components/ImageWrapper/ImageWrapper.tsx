import React, { memo, useRef, useState } from "react";
import classnames from "classnames";

import {
  BorderRadiusProps,
  FlexBasisProps,
  MaxHeightProps,
  OverflowProps
} from "styled-system";

import { handleImageError } from "utils/imageError";
import { useIntersection } from "hooks/useIntersection";
import * as S from "./ImageWrapper.styled";

type Props = {
  source: string;
} & FlexBasisProps &
  BorderRadiusProps &
  OverflowProps &
  MaxHeightProps;

const ImageWrapper = ({ source, ...rest }: Props) => {
  const [isLoaded, setIsLoaded] = useState(false);
  const [isInView, setIsInView] = useState(false);
  const imgRef = useRef();
  useIntersection(imgRef, () => {
    setIsInView(true);
  });

  const handleOnLoad = () => {
    setIsLoaded(true);
  };

  return (
    // @ts-ignore
    <S.WrapperImage ref={imgRef} {...rest}>
      {isInView && (
        <S.Image
          className={classnames("image", {
            isLoaded
          })}
          onLoad={handleOnLoad}
          src={source}
          alt={source}
          onError={handleImageError}
        />
      )}
    </S.WrapperImage>
  );
};

export default memo(ImageWrapper);
