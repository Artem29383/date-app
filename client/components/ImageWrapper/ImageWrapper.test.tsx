import { shallow } from 'enzyme';
import ImageWrapper from 'components/ImageWrapper';
import React from 'react';
import { testComponentType } from '@types';

const setUp = (props: any) => shallow(<ImageWrapper {...props} />);

let component: testComponentType;

describe('Test for ImageWrapper', () => {
  beforeEach(() => {
    component = setUp({ source: 'Logo' });
  });

  it('Render ImageWrapper component', () => {
    expect(component).toMatchSnapshot();
  });
});
