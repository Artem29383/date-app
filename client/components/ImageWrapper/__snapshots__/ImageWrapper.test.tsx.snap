// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ImageWrapper Render ImageWrapper component 1`] = `
<styled.div>
  <styled.img
    src="Logo"
  />
</styled.div>
`;
